elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```
@@ -0,0 +1,172 @@
+        }
+
+        File file = new File(args[0]);
+        String absoluteBasePath = DynamicConfigHelpers.formatFilePath(file.getAbsolutePath());
+        log.info("Absolute Path for Model Configs Directory: " + absoluteBasePath);
+
+        if (!file.isDirectory()) {
+            usage();
+            throw new IllegalStateException("Model Configs Directory doesn't exists");
+        }
+
+        readVariableConfig(absoluteBasePath);
+        if (readSecurityConfig(absoluteBasePath)) {
+            validateRoleInSecurityConfig(elideSecurityConfig);
+        }
+        if (readTableConfig(absoluteBasePath)) {
+            validateSqlInTableConfig(elideTableConfig);
+        }
+
+        log.info("Configs Validation Passed!");
+    }
+
+    private static boolean readVariableConfig(String absoluteBasePath) {
+        boolean isVariableConfig = exists(absoluteBasePath + DynamicConfigHelpers.VARIABLE_CONFIG_PATH);
+        try {
+            variables = isVariableConfig ? DynamicConfigHelpers.getVaribalesPojo(absoluteBasePath)
+                    : Collections.<String, Object>emptyMap();
+        } catch (Exception e) {
+            throw new IllegalStateException("Error while parsing variable config at location: " + absoluteBasePath, e);
+        }
+        return isVariableConfig;
+    }
+
+    private static boolean readSecurityConfig(String absoluteBasePath) {
+        boolean isSecurityConfig = exists(absoluteBasePath + DynamicConfigHelpers.SECURITY_CONFIG_PATH);
+        if (isSecurityConfig) {
+            String securityConfigContent = DynamicConfigHelpers
+                    .readConfigFile(new File(absoluteBasePath + DynamicConfigHelpers.SECURITY_CONFIG_PATH));
+            validateConfigForMissingVariables(securityConfigContent, variables);
+            try {
+                elideSecurityConfig = DynamicConfigHelpers.getElideSecurityPojo(absoluteBasePath, variables);
+            } catch (Exception e) {
```

As a general rule, most code should not throw or catch Exception.  Please be explicit.

---

Only checked exceptions need to be mapped to IllegalStateException.  Runtime exceptions don't need to be swallowed here.

---

updated